#!/usr/bin/env python

PACKAGE='husky_trainer'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Name, Type, Level Description, Default, Min, Max
gen.add("lookahead", double_t, 0, "Compute the error according to the position the robot should a bit in the future", 0.2, 0.0, 1.0)
gen.add("lambda_x", double_t, 0, "The proportional gain applied to the error in x", 2.0, 0.0, 5.0)
gen.add("lambda_y", double_t, 0, "The proportional gain applied to the error in y", 1.0, 0.0, 5.0)
gen.add("lambda_t", double_t, 0, "The proportional gain applied to the error in theta", 1.0, 0.0, 5.0)
gen.add("lp_filter_time_const", double_t, 0, "The low-pass filter time constant", 1.0, 0.0, 5.0)
gen.add("max_linear_speed", double_t, 0, "Maximum linear speed.", 1.0, 0.0, 1.0)
gen.add("min_linear_speed", double_t, 0, "Min linear speed.", -1.0, -1.0, 0.0)
gen.add("max_angular_speed", double_t, 0, "Maximum angular speed.", 1.0, 0.0, 1.0)
gen.add("min_angular_speed", double_t, 0, "Minimum angular speed.", -1.0, -1.0, 0.0)

exit(gen.generate(PACKAGE, "repeat", "Repeat"))
